File: ClientApp/src/app/Models/questionProfile.ts
import { AnswerDetailsView } from "./AnswerDetailsView";
import { Category } from "./category";
import { UserDetailsView } from "./userDetailsView";

export interface QuestionProfile{
    id : number
    views : number
    createdOn : Date
    votes : number
    numberOfAnswers : number
    headline : string
    description : string
    user : UserDetailsView
    isSolved : boolean 
    category : Category 
    answers : Array<AnswerDetailsView> 
    questionTime : string;
    
}

File: Modals/CoreModals/QuestionProfile.cs
using Modals.DataModels;
using Modals.DataViewModels;
using System;
using System.Collections.Generic;
using System.Text;

namespace Modals.CoreModals
{
    public class QuestionProfile
    {

        public int Id { get; set; }
        public int Views { get; set; }
        public DateTime CreatedOn { get; set; }
        public int Votes { get; set; }
        public int NumberOfAnswers { get; set; }
        public string Headline { get; set; }
        public string Description { get; set; }
        public UserDetailsView User { get; set; }
        public bool IsSolved { get; set; }
        public Category Category { get; set; }
        public List<AnswerDetailsView> Answers { get; set; }
    }
}
